### ISSUE TITLE
Missing resources

### ISSUE THREAD

**Issue by stormmore** [kubeapps/kubeapps#632]
So during developing a chart and testing deploys I have gotten into a situation where Kubeapps doesn't display any of the resource:

![image](https://user-images.githubusercontent.com/23155715/45503311-53b31980-b73b-11e8-9e58-5ca6a70752a6.png)

I suspect I am doing something stupid with labels but not 100% sure.

Here is the chart:

[gitea-0.2.0.tar.gz](https://github.com/kubeapps/kubeapps/files/2380340/gitea-0.2.0.tar.gz)

It happens even with the default values.

Here is my `helm version`:

```
Client: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0", GitCommit:"9ad53aac42165a5fadc6c87be0dea6b115f93090", GitTreeState:"clean"}
```

**Comment by migmartri** [comment]
I've been looking into this and this is what've found:

TLDR; The deployment has a duplicated `chart` label:

### Root cause
The app is actually crashing complaining about a duplicated key.

![selection_703](https://user-images.githubusercontent.com/24523/45508410-a0055600-b749-11e8-86e6-bae447806b31.png)

Looking into the tags in the rendered template I noticed that it was the deployment the one that had duplicated labels.

```yaml
helm get manifest miguel-test-git | grep   'chart: "gitea-0.2.0"' -C 5

kind: Deployment
metadata:
  name: miguel-test-git-gitea
  labels:
    app: gitea
    chart: "gitea-0.2.0"
    heritage: "Tiller"
    release: "miguel-test-git"
    chart: "gitea-0.2.0"
```

and indeed the chart template includes the duplicated template

```yaml
metadata:
  name: {{ include "gitea.fullname" . }}
  labels:
    app: {{ include "gitea.name" . }}
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
    heritage: "{{ .Release.Service }}"
    release: "{{ .Release.Name }}"
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```

Removing the duplicating key and upgrading the release fixed the issue:

![selection_704](https://user-images.githubusercontent.com/24523/45508638-26ba3300-b74a-11e8-8fb1-bab9055242a9.png)

### Action items

* Our error handling here seems deficient and misleading, we need to improve it. https://github.com/kubeapps/kubeapps/issues/635
* About this specific case, apparently helm is OK with this duplication issue so we might also want to relax this rule a little bit. I am not convinced about this though as long as we show the user enough info about what happened by implementing #1 
* Potentially the helm linter should detect this issue 

``` bash
$ helm lint .
==> Linting .
Lint OK
```

**Comment by migmartri** [comment]
@stormmore I have created two follow up tasks related to the action items, so if you do not mind I am going to go ahead and close this issue. Let me know if you want to keep it open for any reason.

Thanks for reporting the issue!
